flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -323,9 +323,8 @@ public final void invoke() throws Exception {
 
 			CheckpointExceptionHandlerFactory cpExceptionHandlerFactory = createCheckpointExceptionHandlerFactory();
 
-			synchronousCheckpointExceptionHandler = cpExceptionHandlerFactory.createCheckpointExceptionHandler(
-				getExecutionConfig().isFailTaskOnCheckpointError(),
-				getEnvironment());
+			synchronousCheckpointExceptionHandler = cpExceptionHandlerFactory
```

Comment:
I wonder if we still need the sync and async exception handler, factoy, etc ... because there is only a single implementation left now and that could be inlined again.

Reply:
Agree, there is no more benefit to distinguish these two exception handlers. Already removed the `asynchronousCheckpointExceptionHandler` in `StreamTask`.